### PR TITLE
Adding some filename sanitization to example's upload.php

### PR DESCRIPTION
**Pull request by fidian**
First, get just the filename portion of the name.  One could send
`../../../../../../etc/passwd` or something crazy and we just want the
`passwd` portion.

Next, eliminate all leading periods.  That way we don't create
`.htaccess` files by accident.

Change all unsafe characters to underscore.

Finally, if that produces an empty string, we'll just make up our own
filename.

### CODE REVIEW

**Comment by jayarjo** [comment]
Please check: https://github.com/moxiecode/plupload/pull/1325.

**Comment by fidian** [comment]
Why was this pull request closed?

I am a bit confused.  I see a project with an example PHP upload handler.  I figure examples should illustrate best practices and should be improved when they possibly have problems.

I read a ticket that said the example should not be used.  There's no documentation that I found indicating that the examples should not be used.  The getting started wiki pages make no mention of using other projects that already exist.

I feel that the pull request should be reopened or some other commits should eliminate the upload handler examples as well as update documentation to guide new users to alternate projects.

**Comment by jayarjo** [comment]
Example must be as simple as possible. And I do not mean that our examples are the best. I'd say the opposite. But they mainly demonstrate must-have principles (more or less). What you suggest is subjective way of sanitizing file names, many users won't like the way it treats unicode file names for example...  (is `g` modifier even valid in php?).

The referenced thread has a link to a project that proposes more customizable backend code.

File: examples/upload.php [hunk 1]
```diff
@@ -58,6 +58,15 @@
 	$fileName = uniqid("file_");
 }
 
+// Sanitize file name
+$fileName = basename($filename);
```
Comment (jayarjo) [inline examples/upload.php]: This one makes sense.

**Pull request closed without merging**